packages/react-client/src/ReactFlightReplyClient.js

```diff
@@ -63,10 +63,13 @@ export type EncodeFormActionCallback = <A>(
 
 export type ServerReferenceId = any;
 
-const knownServerReferences: WeakMap<
-  Function,
-  {id: ServerReferenceId, bound: null | Thenable<Array<any>>},
-> = new WeakMap();
+type ServerReferenceMetaData = {
```

Comment:
The terminology here is a bit unclear but `ClientReferenceMetadata` is actually the equivalent of `ServerReferenceId`. I.e. excluding the bound arguments. We should really have bound client references too.

Originally I intended to refer to the Metadata + Bound arguments as a "Reference" inside the payload itself because it's overloaded both in terms of the first class object exposed to user space and references to various IDs in the payload format.

This pairing of bound arguments + the id/metadata should probably have its own name. Maybe ServerReferenceClosure?